genie-core/src/main/java/com/netflix/genie/core/jpa/entities/AuditEntity.java

```diff
@@ -0,0 +1,115 @@
+/*
+ *
+ *  Copyright 2015 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.core.jpa.entities;
+
+import com.netflix.genie.core.jpa.entities.projections.AuditProjection;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.ToString;
+
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.PrePersist;
+import javax.persistence.PreUpdate;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+import javax.persistence.Version;
+import java.util.Date;
+
+/**
+ * Abstract class to support basic columns for all entities for genie.
+ *
+ * @author tgianos
+ */
+@Getter
+@ToString(callSuper = true)
+@MappedSuperclass
+public class AuditEntity extends IdEntity implements AuditProjection {
+
+    @Basic(optional = false)
+    @Column(name = "created", nullable = false, updatable = false)
+    @Temporal(TemporalType.TIMESTAMP)
+    // TODO: Test out this instead of all the pre-persist stuff
+//    @CreatedDate
+    private Date created = new Date();
+
+    @Basic(optional = false)
+    @Column(name = "updated", nullable = false)
+    @Temporal(TemporalType.TIMESTAMP)
+    // TODO: Test out this instead of all the pre-persist stuff
+//    @LastModifiedDate
+    private Date updated = new Date();
+
+    @Version
+    @Column(name = "entity_version", nullable = false)
+    @Getter(AccessLevel.NONE)
+    private Integer entityVersion;
+
+    /**
+     * Updates the created and updated timestamps to be creation time.
+     */
+    @PrePersist
+    protected void onCreateBaseEntity() {
+        final Date date = new Date();
+        this.updated = date;
+        this.created = date;
```

Comment:
No. The method is annotated with `@PrePersist` which is only called the first time the entity is saved to the database. Also the created field is annotated with `@Column(name = "created", nullable = false, updatable = false)` and the `updatable=false` makes it so that the value is never included in update queries.